lib/stdlib/src/supervisor.erl

```diff
@@ -1184,36 +1197,35 @@ count_child(#child{pid = Pid, child_type = supervisor},
 %%% check it's inbox before trying again.
 -doc false.
 -spec handle_cast({try_again_restart, child_id() | {'restarting',pid()}}, state()) ->
-			 {'noreply', state()} | {stop, shutdown, state()}.
+			 {'noreply', state(), gen_server:action()} | {stop, shutdown, state()}.
 
 handle_cast({try_again_restart,TryAgainId}, State) ->
     case find_child_and_args(TryAgainId, State) of
 	{ok, Child = #child{pid=?restarting(_)}} ->
 	    case restart(Child,State) of
 		{ok, State1} ->
-		    {noreply, State1, State1#state.hibernate_after};
+		    {noreply, State1, hibernate_after_action(State)};
 		{shutdown, State1} ->
 		    {stop, shutdown, State1}
 	    end;
 	_ ->
-	    {noreply,State, State#state.hibernate_after}
+	    {noreply,State, hibernate_after_action(State)}
     end.
 
 %%
 %% Take care of terminated children.
 %%
 -doc false.
 -spec handle_info(term(), state()) ->
-        {'noreply', state()} | {'stop', 'shutdown', state()}.
+        {'noreply', state(), gen_server:action()} | {'stop', 'shutdown', state()}.
 
-handle_info(timeout, State) ->
-    logger:notice("Hibernating supervisor ~p", [self()]),
+handle_info(HRef, #state{hibernate_after_ref = HRef} = State) ->
```

Comment:
```suggestion
handle_info({HRef, hibernate}, #state{hibernate_after_ref = HRef} = State) ->
```
This (together with a complementary change in `hibernate_after_action`) makes it more clear what this is about, especially if we think of using not a specialized `hibernate_after` timeout tag but a general "supervisor message to itself" tag as I suggested in another comment.